doc/specs/#6049 - Font config objects.md

```diff
@@ -0,0 +1,98 @@
+
+## Solution Design
+
+The implementation design for font configuration objects will be very similar to the way appearance configuration objects were implemented ([reference](https://github.com/microsoft/terminal/pull/8345)). There will be a default font object, that will contain settings for the rendering of font in the default case. Beyond that, a number of optional font objects can be created for various use cases such as the rendering of bold font or italic font. These optional font objects will have a smaller number of allowed parameters, and will inherit from the default font object (i.e. any parameters that they do not define will be taken from the default font object).
+
+### Allowed parameters and allowed non-default font configurations
+
+As of now, Windows Terminal supports 3 font-related settings: `fontFace`, `fontWeight` and `fontSize`. The default font object will be allowed to define all 3 of these settings, whereas the non-default font objects will only be allowed to define `fontFace` and `fontWeight`. We do not allow the non-default font objects to define a `fontSize` as we do not want there to be resizes as the text switches between different types of font like bold/italic.
+
+In the initial stages of this feature, we will only allow two non-default font configurations: `italicFont` and `boldFont`.
+
+### Inheritance
+
+The font configuration class will be implemented such that it allows for inheritance between objects of the class. As described earlier, within the same profile, non-default font configuration objects will inherit from the default font configuration object.
+
+*Between* profiles, each profile's default font configuration object will inherit from the default font configuration object defined in `profiles.defaults`. As for the non-default font configuration objects, these are treated as a single setting. I.e. a profile's `italicFont` font configuration object does *not* inherit from the `italicFont` object defined in `profiles.defaults`, it is simply its own setting. The `italicFont` configuration object defined in `profiles.defaults` only applies to profiles that do not define their own `italicFont`.
+
+This design is in line with the way we handle `defaultAppearance` and `unfocusedAppearance`, and was deliberately chosen so we maintain consistency across these different classes.
+
+## UI/UX Design
+
+Users will be able to add font configurations to their profiles that will look something like this:
+
+```
+"font": // this is the default
+{
+    "fontFace": "Cascadia Mono",
+    "fontWeight": "Normal",
+    "fontSize": 12
+},
+"boldFont":
+{
+    "fontFace": "Consolas",
+    "fontWeight": "Extra-Bold"
+}

```

Comment:
You know what, let me test that out and get back to you

Reply:
Tested it, and short answer: yes we can have multiple fonts in the same text buffer

Long answer: there's funky stuff that goes on because of the different fonts not necessarily having the same glyph size and so the spacing becomes weird - we _should_ be able to do some creative scaling to avoid this problem though